File: 0485_findMaxConsecutiveOnes.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        result = 0
        count = 0
        for num in nums:
            if num == 1:
                count += 1
                result = max(result, count)
            else:
                count = 0
        return result

# Alternative solution

class Solution:
    def findMaxConsecutiveOnes(self, nums):
        left = -1
        maxOnes = 0
        for right, num in enumerate(nums):
            if num == 1:
                maxOnes = max(maxOnes, right - left)
            else:
                left = right
        return maxOnes

File: 0001-0500/0485_findMaxConsecutiveOnes.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int max = 0;
        int left = 0;
        int right = 0;

        while (right < nums.length) {
            if (nums[right] == 1) {
                right++;
                max = Math.max(max, right - left);
            } else {
                right++;
                left = right;
            }
        }

        max = Math.max(max, right - left);
        return max;
    }
}
